core/src/autogluon/core/models/greedy_ensemble/cmaes_weights.py

```diff
@@ -0,0 +1,448 @@
+        internal_solutions = solutions
+
+        # -- Normalize if enabled
+        #if True:
+            #internal_solutions = np.array([self._normalize_weights(s) for s in internal_solutions])
+
+        if len(internal_solutions) > 1:
+            losses = np.apply_along_axis(
+                partial(self.evaluate_single_solution, *func_args),
+                axis=1,
+                arr=internal_solutions,
+            )
+        else:
+            losses = np.array([self.evaluate_single_solution(*func_args, internal_solutions[0])])
+
+        return losses
+    
+    def evaluate_single_solution(
+        self,
+        predictions,
+        labels,
+        score_metric,
+        normalize_predict_proba,
+        weight_vector,
+    ):
+        """
+        This method calculates the loss for a batch of candidate solutions by applying the evaluate_single_solution 
+        method to each solution. It returns the calculates losses. 
+
+        Parameters
+        ----------
+        predictions: np.ndarray
+            Prediction probabilities from multiple models
+        labels: np.ndarray
+            Ground truth labels 
+        score_metric: 
+            Evaluation metric to evaluate the predicted labels against the ground truth labels
+        normalize_predict_proba:
+            Option not implemented
+        weight_vector: np.ndarray
+            Weighted ensemble vector
+
+        Returns
+        -------
+        Ensemble performance evaluated using passed in score metric
+        """
+        y_pred_ensemble = AbstractWeightedEnsemble.weight_pred_probas(
+            predictions,
+            weight_vector,
+        )
+        return score_metric(labels, y_pred_ensemble)
+
+    def _calculate_regret(self, y_true: np.ndarray, y_pred_proba: np.ndarray, metric, sample_weight=None):
+        if metric.needs_pred or metric.needs_quantile:
+            preds = get_pred_from_proba(y_pred_proba=y_pred_proba, problem_type=self.problem_type)
+        else:
+            preds = y_pred_proba
+        score = compute_weighted_metric(y_true, preds, metric, sample_weight, quantile_levels=self.quantile_levels)
+        return metric._optimum - score
+
+    def _calculate_weights(self):
+        ensemble_members = Counter(self.indices_).most_common()
+        weights = np.zeros((self.num_input_models_,), dtype=float)
+        for ensemble_member in ensemble_members:
+            weight = float(ensemble_member[1]) / self.ensemble_size
+            weights[ensemble_member[0]] = weight
+
+        if np.sum(weights) < 1:
+            weights = weights / np.sum(weights)
+
+        self.weights_ = weights
+
+
+class SimpleWeightedEnsemble(AbstractWeightedEnsemble):
+    """Predefined user-weights ensemble"""
+
+    def __init__(self, weights, problem_type, **kwargs):
+        self.weights_ = weights
+        self.problem_type = problem_type
+
+    @property
+    def ensemble_size(self):
+        return len(self.weights_)
```

Comment:
Why are we duplicating code that exists in https://github.com/alee4977/autogluon/blob/731e61a7be08e2f7dc8b2aa81873d649ff165cb2/core/src/autogluon/core/models/greedy_ensemble/ensemble_selection.py ?